src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,549 @@
+from logging import exception
+import os
+import json
+import datetime
+from subprocess import Popen, PIPE, run, STDOUT, call, DEVNULL
+from base64 import b64encode, b64decode
+from unicodedata import name
+from azure.core.exceptions import ClientAuthenticationError
+import shutil
+from _thread import interrupt_main
+from psutil import process_iter, NoSuchProcess, AccessDenied, ZombieProcess, net_connections
+from knack.util import CLIError
+from knack.log import get_logger
+from knack.prompting import prompt_y_n
+from knack.prompting import NoTTYException
+from azure.cli.core.commands.client_factory import get_subscription_id
+from azure.cli.core._profile import Profile
+from azure.cli.core.util import sdk_no_wait
+from azure.cli.core import telemetry
+from azure.cli.core.azclierror import ManualInterrupt, InvalidArgumentValueError, UnclassifiedUserFault, CLIInternalError, FileOperationError, ClientRequestError, DeploymentError, ValidationError, ArgumentUsageError, MutuallyExclusiveArgumentError, RequiredArgumentMissingError, ResourceNotFoundError
+from Crypto.IO import PEM
+from Crypto.PublicKey import RSA
+from Crypto.Util import asn1
+from azext_connectedk8s._client_factory import _graph_client_factory
+from azext_connectedk8s._client_factory import cf_resource_groups
+from azext_connectedk8s._client_factory import _resource_client_factory
+from azext_connectedk8s._client_factory import _resource_providers_client
+from azext_connectedk8s._client_factory import get_graph_client_service_principals
+import azext_connectedk8s._constants as consts
+import azext_connectedk8s.custom as custom
+from glob import glob
+from .vendored_sdks.models import ConnectedCluster, ConnectedClusterIdentity, ListClusterUserCredentialProperties
+from threading import Timer, Thread
+from azure.cli.core import get_default_cli
+logger = get_logger(__name__)
+# pylint:disable=unused-argument
+# pylint: disable=too-many-locals
+# pylint: disable=too-many-branches
+# pylint: disable=too-many-statements
+# pylint: disable=line-too-long
+
+def create_folder_diagnosticlogs(time_stamp, storage_space_available):
+
+    home_dir = os.path.expanduser( '~' )
+    filepath = os.path.join( home_dir, 'diagnostic_logs' )
+    filepath_with_timestamp = filepath
+    try:
+        
+        # Creating the Diagnoser Folder and adding it if its not already present
+        try:
+            os.mkdir(filepath)
+        except FileExistsError:
+            pass
+        
+        # Creating Subfolder with the given timestamp to store all the logs
+        filepath_with_timestamp = os.path.join(filepath, time_stamp) 
+        try:
+            os.mkdir(filepath_with_timestamp)
+        except FileExistsError:
+            pass
+        return filepath_with_timestamp, storage_space_available
+
+    except OSError as e:
+        storage_space_available = False
+        shutil.rmtree(filepath_with_timestamp, ignore_errors=False, onerror=None)
+
+    except Exception as e:
+        print(e)
```

Comment:
Let's try to print a more descriptive error here, like - An exception has occurred while creating the diagnostic logs folder in your local machine : e

Reply:
Use the logger to display errors instead of print statements

Reply:
Changed all the print(e) after exceptions into logger.warning with an informative message.